src/test/java/build/buildfarm/instance/stub/ByteStringIteratorInputStreamTest.java

```diff
@@ -116,4 +118,51 @@ public void readWithOffsetAndLengthSpanningChunks() throws IOException {
     assertThat(in.read()).isEqualTo(3);
     assertThat(in.read()).isEqualTo(-1);
   }
+
+  @Test
+  public void iteratorAtBeginningHasAvailable() throws IOException {
+    InputStream in = new ByteStringIteratorInputStream(ImmutableList.<ByteString>of().iterator());
+    assertThat(in.available()).isEqualTo(0);
+  }
+
+  @Test
+  public void iteratorAtEndIsUnavailable() throws IOException {
+    InputStream in = new ByteStringIteratorInputStream(ImmutableList.<ByteString>of().iterator());
+    assertThat(in.available()).isEqualTo(0);
+  }
+
+  @Rule
+  public final ExpectedException exception = ExpectedException.none();
+
+  @Test
+  public void closedStreamAvailableThrowsIOException() throws IOException {
+    InputStream in = new ByteStringIteratorInputStream(
+        ImmutableList.<ByteString>of(
+            ByteString.copyFromUtf8("Hello, World")).iterator());
+    in.close();
+    exception.expect(IOException.class);
```

Comment:
nit: For your use-cases here, the @Test(expected = IOException.class) idiom would be simpler, I think.
Also, you can use fail("excepted exception") instead of assert true = false